extension/api.js

```diff
@@ -112,15 +112,27 @@ const lostEngines = new Map ([
 // we will try again later.
 let tryagain = false;
 
-function finish(result = false) {
-  Services.prefs.setBoolPref(RUN_ONCE_PREF, result);
+function finish(result = false, event, reason, id = null) {
```

Comment:
All parameters here appear to be mandatory, for this logic to make sense. Let's remove the default values.

Reply:
id is only included if the panel is shown, so it will remain with a default.